File: Sdks/AdminApi.Sdk/src/Endpoints/Identities/Types/IndividualQuota.cs
namespace Backbone.AdminApi.Sdk.Endpoints.Identities.Types;

public class IndividualQuota
{
    public required string Id { get; set; }
    public required Metric Metric { get; set; }
    public required int Max { get; set; }
    public required string Period { get; set; }
}

File: Applications/AdminUi/packages/admin_api_types/lib/src/identities/individual_quota.dart
import 'package:json_annotation/json_annotation.dart';

import '../metrics/metrics.dart';

part 'individual_quota.g.dart';

@JsonSerializable()
class IndividualQuota {
  final String id;
  final Metric metric;
  final int max;
  final String period;

  IndividualQuota({required this.id, required this.metric, required this.max, required this.period});

  factory IndividualQuota.fromJson(dynamic json) => _$IndividualQuotaFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$IndividualQuotaToJson(this);
}
